docs/index.md

```diff
@@ -1,53 +1,142 @@
+
+1. Starts and runs your services in an order that respects the dependencies you
+   define.
+
+1. Provides containers the environment variables that they need to run.
+
+The simplest way to use Maestro with your application is to write a Python
+script that connects them together. Maestro provides a set of [Guest
 functions](guest-functions.md) that know how to grok this environment
-information.
+information to make this easier.
+
+## Quickstart
+
+Let's get a quick app running with Flask, Redis, and Maestro. Make sure to
+install Maestro if you haven't already.
+
+```
+$ pip install maestro-ng
+```
+
+The first thing we'll do is write our `maestro.yaml` file which tells Maestro
+how to start and run our services:
+
+```yaml
+name: demo
+
+ships:
+  localhost:
+    ip: 192.168.10.2
+
+services:
+  redis:
+    image: redis
+    instances:
+      redis-1:
+        ship: localhost
+        ports:
+            redis: 6379
+        lifecycle:
+            running: [{type: tcp, port: redis}]
+  web:
+    image: ceasar/web
+    requires: [redis]
+    instances:
+      web-1:
+        ship: localhost
+        ports:
+            http: 5000
+        lifecycle:
+            running: [{type: tcp, port: http}]
+```
 
-----
+So what exactly does this do?
 
-Let's first look at how environments and services are described, then
-we'll discuss what information Maestro passes down to the containers
-through their environment.
+1. The `name` section simply gives a name to our environment.
+
+1. The `ships` section tells Maestro about the machines that we want our Docker
+   containers to run on. In this case, we only have one machine, our computer.
+
+1. The `services` section tells Maestro about the services we want to run. In
+   this case, we have two: `redis`, a key-value store, and `web`, a Flask
+   application. We want both services to run on `localhost`, and we want to
+   expose their default ports (6379 for Redis and 5000 to Flask) so that Flask
+   can reach Redis, and we can reach Flask. We also tell Maestro that `web`
+   depends on `redis`, so that Maestro doesn't start `web` until `redis` is
+   running. The `lifecycle` sections tell Maestro how to check if a service is
+   running successfully. In this case, our services are running correctly if
+   Maestro can reach them at their exposed ports.
+
+Next, we'll create our Flask app in ``app.py``, which simply print the number
+of times the index has been visited:
+
+```python
+import os
+
+from flask import Flask
+from maestro.guestutils import get_node_list, get_port
+from redis import Redis
+
+
+app = Flask(__name__)
+redis_host, redis_port = get_node_list('redis', ports=['redis'], minimum=1)[0].split(':')
+redis = Redis(host=redis_host, port=int(redis_port))
+
+@app.route('/')
+def hello():
+    redis.incr('hits')
+    return 'Hello World! I have been seen %s times.' % redis.get('hits')
+
+if __name__ == "__main__":
+    app.run(host="0.0.0.0", port=get_port('http'), debug=True)
+```
```

Comment:
Not sure if it's okay to copy the example from Fig so blatantly. Maybe we should make a slightly different example.